types/block.go

```diff
@@ -142,3 +142,23 @@ func (b *Block) Score() uint64 {
 func (b *Block) Equals(other *Block) bool {
 	return b.Cid().Equals(other.Cid())
 }
+
+// DataToSign returns the block's bytes without the blocksig for signature
+// creating and verification
+func (b *Block) DataToSign() []byte {
```

Comment:
nit: Since we also pass this to verification methods, maybe call this `SignatureData`? A similar pattern is used for payment vouchers.

Reply:
Alternatively, something like the below would also be nice. 
```go
func (b *Block) Sign(address address.Address, signer Signer) error {}
func (b *Block) ValidSignature(addr address.Address, verifier Recoverer) bool {}
```

Reply:
Prefer the former: `Block` shouldn't depend on `Signer` and `Recoverer`.